### ISSUE TITLE
Use default path /test/** only if src is not defined

### ISSUE THREAD

**Issue by vitalets** [tandrewnichols/grunt-only#1]
hi, I want to use grunt-only only with files staged for commit. For that I'm updating src dynamically with staged files. But when I get empty src, grunt-only checks all files in default path `test/**`.
I suggest to add condition here to apply default path only if src is not defined:
https://github.com/tandrewnichols/grunt-only/blob/master/tasks/only.js#L21

**Comment by tandrewnichols** [comment]
Yeah, I think that makes sense. Or at least, the use case is valid, and there should be a way to set src dynamically.

**Comment by tandrewnichols** [comment]
@vitalets, I went a different way with this. Have a look at #2, and let me know if there's any reason that doesn't satisfy your use case. Thanks!

**Comment by tandrewnichols** [comment]
Actually, now that I've done that . . . maybe I don't understand your issue. If I have a grunt setup like this:

```
// Omitting more here

grunt.initConfig({
  only: {
    tests: {
      src: []
    }
  }
});

grunt.registerTask('dynamic', function() {
  grunt.config.set('only.tests.src', ['test/foo.js']);
});
```

And I run `grunt dynamic only`, the right files get run (only test/foo.js and nothing else). This works both with no `src` property at all and with `src` set to an empty array. Maybe I need a more specific example of how this is failing for you.

**Comment by vitalets** [comment]
hi, 
consider this:

``` js
grunt.registerTask('dynamic', function() {
  var changedFiles = getChangedFiles(); // if empty array here, grunt-only will run over test/**
  grunt.config.set('only.tests.src', changedFiles);
});
```

**Comment by tandrewnichols** [comment]
Oh, I see. So you're talking specifically about the case when src is (or _could be_) empty. Your original suggestion makes a lot of sense now.

**Comment by tandrewnichols** [comment]
Published in v1.0.4. Thanks for opening the issue!
